### DISCUSSION TITLE
Is it possible to remove a ghostty icon from a dock on MacOS?

### DISCUSSION THREAD

**Discussion by frolvanya** [ghostty-org/ghostty#3810] [Q&A] [answered]
<img width="65" alt="image" src="https://github.com/user-attachments/assets/48ebbe44-fa1c-4e8d-85a7-6fb7ffbe4bd0" />

I'm using ghostty primarily in quick-terminal mode, so I don't want to see an icon in the dock

**Comment by jmckillican** [comment]
I don't believe there's a way through Ghostty options but you can use this like on any other macOS app to hide the dock icon.


> 1. quit the application you want to hide from the dock
> 2. Go to your applications folder and right control-click on the app you want to hide from the dock
> 3. Click on "Show contents"
> 4. Open the "Contents" folder
> 5. Open the file Info.plist with TextEdit (right click -> "Open with...")
> 6. At the bottom of this preference file are 2 lines which say:
> ```
> </dict>
> </plist>
> ```
> 7. Just before these two lines type the following:
> ```
> <key>LSUIElement</key>
> <true/>
> ```
> 8. Save and close the file
> 9. Re-open the app - the dock icon should no longer be visible

**Reply by frolvanya** [reply]
Yeah, that's the first thing that came to my mind, but this messes up my key bind for some reason. I have this line in my config: `keybind = global:cmd+`=toggle_quick_terminal`. Without LSUIElement it triggers everywhere, but when I add LSUIElement it triggers only one time and then I need to be focused on a regular ghostty window in order for it to work. Do you know what can cause this issue?

**Comment by univeous** [comment]
I want to add some information here. [Raycast](https://www.raycast.com/) does not display an icon in the dock by default, but it still responds to hotkeys normally; when a specific window, such as the raycast settings window, is opened, the icon will appear in the dock and cmd+tab.

It would be great if Ghostty could implement a similar mechanism, such as hiding from the dock when no terminal window is open, and still respond to global hotkeys, used for `toggle_quick_terminal` and so on;

When a terminal window is open, the icon should appear in the dock (and cmd+tab).

FYI: https://stackoverflow.com/questions/12897214/transform-lsuielement-to-foreground-application

**Reply by frolvanya** [reply]
> When a terminal window is open, the icon should appear in the dock (and cmd+tab).

Not completely agree with you on this one, I'd like to see 3 options for a new flag (e.g `macos-icon-visibility`):
- always
- never
- when-open

Or something like that. But yeah, I generally support this idea

P.S: Hmm, Raycast also uses `<key>LSUIElement</key>`, but for some reason it doesn't mess up shortcuts

**Comment by crabique** [comment]
I would love to have that feature for the regular terminal window as well, not just the quick one.

Also, common problem with this feature in [similar programs](https://gitlab.com/gnachman/iterm2/-/issues/10973) is that this breaks Apple's "remember input layout" behavior, if the user has multiple input sources configured, the terminal will continue using the one that had been active before the shortcut was pressed.

**Comment by mitchellh** [comment, marked as answer]
Approved #4538

**Comment by switz** [comment]
Ghostty shows up for me in alt tab despite having this enabled, I presume because I have it manually in the dock. Anyway to have this niche setup work?

**Reply by Write** [reply]
> Ghostty shows up for me in alt tab despite having this enabled, I presume because I have it manually in the dock. Anyway to have this niche setup work?

Honstely, just use Alt Tab app.

**Reply by switz** [reply]
I don't think the solution to a problem should be "install another app". This was possible in iterm, so it should be possible here.
